### PR TITLE
[css-animations-2] Add testing coverage for `animation-composition`

### PR DESCRIPTION
**Pull request by graouts**
There is no test coverage for the [`animation-composition`](https://drafts.csswg.org/css-animations-2/#animation-composition) property. At the moment that property is not implemented by any browser engine that I know of, but WebKit plans to implement it soon. It should be straightforward for Gecko and Blink to add support as well considering they have long-established support for Web Animations composite operations and this CSS property uses existing patterns to expose Web Animations features in CSS Animations.

Adding support for this property in WebKit is tracked by [WebKit bug 232086](https://bugs.webkit.org/show_bug.cgi?id=232086).

I believe these changes are exhaustive and comparable to other `animation-` CSS properties for parsing support. I'll note a couple of more specific things:

- `css/css-animations/CSSAnimation-effect.tentative.html`: check that setting `animation-composition` sets the `composite` property on the effect itself,
- `css/css-animations/KeyframeEffect-getKeyframes.tentative.html`: check that `animation-composition` is accounted for when set on keyframes within a `@keyframes` rule to de-duplicate keys, along with the offset and the timing function.

The rest of the changes are pretty standard.

### CODE REVIEW

File: css/css-animations/KeyframeEffect-getKeyframes.tentative.html [hunk 1]
```diff
@@ -47,6 +47,18 @@
   to   { color: rgb(255, 255, 255); }
 }
 
+@keyframes anim-simple-composite {
+  from { color: rgb(0, 0, 0); animation-composition: replace; }
+  50%  { color: rgb(0, 0, 255);  animation-composition: add; }
```
Comment (heycam) [inline css/css-animations/KeyframeEffect-getKeyframes.tentative.html:52]: Nit: one too many spaces before `animation-composition`.

File: css/css-animations/parsing/animation-computed.html [hunk 2]
```diff
@@ -15,22 +15,23 @@
...
-test_computed_value("animation", "cubic-bezier(0, -2, 1, 3)", "0s cubic-bezier(0, -2, 1, 3) 0s 1 normal none running none");
-test_computed_value("animation", "1s -3s", "1s ease -3s 1 normal none running none");
-test_computed_value("animation", "4", "0s ease 0s 4 normal none running none");
-test_computed_value("animation", "reverse", "0s ease 0s 1 reverse none running none");
-test_computed_value("animation", "both", "0s ease 0s 1 normal both running none");
-test_computed_value("animation", "paused", "0s ease 0s 1 normal none paused none");
-test_computed_value("animation", "none", "0s ease 0s 1 normal none running none");
-test_computed_value("animation", "anim", "0s ease 0s 1 normal none running anim");
+// <single-animation-composition> || [ none | <keyframes-name> ]
+test_computed_value("animation", "1s", "1s ease 0s 1 normal none running replace none");
```
Comment (heycam) [inline css/css-animations/parsing/animation-computed.html:19]: Is this the usual way we update WPTs for new longhand components of existing shorthands? This change will make the test (as of right now) fail in all engines. Not sure if we want to go to the extent of something like:

```js
let composition = CSS.supports("animation-composition", "replace") ? " replace": "";
test_computed_value("animation", "1s", `1s ease 0s 1 normal none running${composition} none`);
...
```

File: css/css-animations/KeyframeEffect-getKeyframes.tentative.html [hunk 3]
```diff
@@ -113,6 +125,28 @@
   to   { margin-top: 20px; margin-right: 20px; margin-bottom: 20px; }
 }
 
+@keyframes anim-merge-offset-and-composite {
```
Comment (heycam) [inline css/css-animations/KeyframeEffect-getKeyframes.tentative.html:128]: I think this test makes sense although I couldn't find where in the specs it's defined that animation-composition participates in the keyframe merging behavior like animation-timing-function does.

**Review by heycam — APPROVED** [review]
r=me though see the comment I left in w3c/csswg-drafts#6930 about the shorthand.

**Comment by graouts** [comment]
> r=me though see the comment I left in [w3c/csswg-drafts#6930](https://github.com/w3c/csswg-drafts/pull/6930) about the shorthand.

I will update this PR to not include the shorthand stuff and wait for this issue to be resolved.

File: css/css-animations/KeyframeEffect-getKeyframes.tentative.html (hunk 3 above)
Comment (graouts) [reply css/css-animations/KeyframeEffect-getKeyframes.tentative.html:128]: I've filed https://github.com/w3c/csswg-drafts/issues/6935.

**Comment by graouts** [comment]
> Looks good to me although I wonder if the tests added in `css/css-animations/parsing/` should be marked as `tentative` like the other tests for CSS Animations Level 2?

Good point, I've done that in 7bac5bcb5bab8cbf2fec6d708fd50fa13f2fb591.

**Pull request merged**